lib/autoupdate.ps1

```diff
@@ -318,34 +318,81 @@ function get_version_substitutions([String] $version, [Hashtable] $customMatches
     return $versionVariables
 }
 
-function autoupdate([String] $app, $dir, $json, [String] $version, [Hashtable] $matches) {
-    Write-Host -f DarkCyan "Autoupdating $app"
+function lets_do_autoupdates([String] $app, $json, [String] $version, $substitutions) {
+    $urls = @()
+    $hashes = @()
     $has_changes = $false
     $has_errors = $false
     [Bool]$valid = $true
-    $substitutions = get_version_substitutions $version $matches
 
-    if ($json.url) {
+    for ($i=0; $i -lt $json.autoupdate.url.Length; $i++) {
         # create new url
-        $url   = substitute $json.autoupdate.url $substitutions
-        $valid = $true
+        $url   = substitute $json.autoupdate.url[$i] $substitutions
+        $urls += $url
 
         if($valid) {
             # create hash
-            $hash = get_hash_for_app $app $json.autoupdate.hash $version $url $substitutions
+            if ($json.autoupdate.hash.length -gt $i) {
+                $h = $json.autoupdate.hash[$i]
+            } else {
+                $h = $null
+            }
+            $hash = get_hash_for_app $app $h $version $url $substitutions
+            $hashes += $hash
             if ($null -eq $hash) {
                 $valid = $false
                 Write-Host -f DarkRed "Could not find hash!"
             }
         }
+    }
+
+    # write changes to the json object
+    if ($valid) {
+        $has_changes = $true
+        update_manifest_with_new_version $json $version $urls $hashes
+    } else {
+        $has_errors = $true
+        throw "Could not update $app"
+    }
+
+    return $has_changes, $has_errors
+}
+
+
+function autoupdate([String] $app, $dir, $json, [String] $version, [Hashtable] $matches) {
+    Write-Host -f DarkCyan "Autoupdating $app"
+    $has_changes = $false
+    $has_errors = $false
+    [Bool]$valid = $true
+    $substitutions = get_version_substitutions $version $matches
 
-        # write changes to the json object
-        if ($valid) {
-            $has_changes = $true
-            update_manifest_with_new_version $json $version $url $hash
+    if ($json.url) {
+        if ($json.autoupdate.url -is [System.Array]) {
```

Comment:
This if...else... should be applied to `$json.architecture`'s `ForEach-Object` script too.